### PR TITLE
Saas 17.1 hr fix same user creation mepe

### PR DESCRIPTION
**Pull request by mepe-odoo**
Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/hr/models/hr_employee.py [hunk 1]
```diff
@@ -37,7 +37,7 @@ def _lang_get(self):
     # resource and user
     # required on the resource, make sure required="True" set in the view
     name = fields.Char(string="Employee Name", related='resource_id.name', store=True, readonly=False, tracking=True)
-    user_id = fields.Many2one('res.users', 'User', related='resource_id.user_id', store=True, readonly=False)
+    user_id = fields.Many2one('res.users', 'User', related='resource_id.user_id', store=True, readonly=False, precompute=True)
```
Comment (tivisse) [inline addons/hr/models/hr_employee.py:40]: Could you add a python test as well ?

File: addons/hr/tests/test_hr_employee.py [hunk 2]
```diff
@@ -250,6 +253,34 @@ def test_availability_user_infos_employee(self):
...
+            'name' : 'Test User Company',
+        })
+        self.env['hr.employee'].create({
+            'name': 'Hr Officer - employee',
+            'user_id': self.res_users_hr_officer.id,
+            'company_id': test_company.id,
+        })
+
+        self.res_users_hr_officer.write({'company_ids': test_company.ids, 'company_id': test_company.id})
+        employee_form = Form(self.env['hr.employee'].with_user(self.res_users_hr_officer).with_company(company=test_company.id))
```
Comment (sofiagvaladze) [inline addons/hr/tests/test_hr_employee.py:267]: Maybe write comment like:
```suggestion
        # Try to set the user with existing employee in the company, on a new employee form
        employee_form = Form(self.env['hr.employee'].with_user(self.res_users_hr_officer).with_company(company=test_company.id))
```

File: addons/hr/tests/test_hr_employee.py [hunk 3]
```diff
@@ -250,6 +253,34 @@ def test_availability_user_infos_employee(self):
...
+        employee_form.user_id = self.res_users_hr_officer
+        with mute_logger('odoo.sql_db'), self.assertRaises(UniqueViolation), self.assertRaises(ValidationError), self.cr.savepoint():
+            employee_form.save()
+
+        employee_2 = self.env['hr.employee'].create({
+            'name': 'Hr 2 - employee',
+            'company_id': test_company.id,
+        })
+
+        employee_2_form = Form(employee_2.with_user(self.res_users_hr_officer).with_company(company=test_company.id))
```
Comment (sofiagvaladze) [inline addons/hr/tests/test_hr_employee.py:278]:
```suggestion
        # Try to set the user with existing employee in the company, on another existing employee
        employee_2_form = Form(employee_2.with_user(self.res_users_hr_officer).with_company(company=test_company.id))
```

**Review by sofiagvaladze — CHANGES_REQUESTED** [review]
@mepe-odoo looks good :raised_hands: 

I think it will make tests easier to get with the commits that I added, what do you think?

**Comment by sofiagvaladze** [comment]
robodoo r+

**Pull request closed without merging**